### PR TITLE
Refactor remix init

### PR DESCRIPTION
**Pull request by ahuth**
Refactors to the `remix.init` script. It runs when people generate a new app.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ FROM base as deps
 
 WORKDIR /myapp
 
-ADD package.json ./
+ADD package.json package-lock.json ./
```
Comment (ahuth) [inline Dockerfile:12]: `remix.init` adds `package-lock.json` here anyway.

File: remix.init/index.js [hunk 2]
```diff
@@ -1,64 +1,39 @@
...
+
+/**
+ * Prepare the generated repo for development by doing things like:
+ * - Replacing "edu-stack" in various files with the app's actual name
+ * - Changing commands to use `npm` or `yarn` (whichever was used to generate the repo)
+ * - etc.
+ *
+ * @param {{isTypeScript: boolean, packageManager: string, rootDirectory: string}} param0
+ */
+module.exports = async function main({
```
Comment (ahuth) [inline remix.init/index.js:15]: Re-arranged things so the `main` function is at the top, and the helper functions are below it. Makes it easier to come in to the file and see what the whole point of it is.

File: remix.init/index.js [hunk 3]
```diff
@@ -81,7 +56,7 @@ const main = async ({isTypeScript, packageManager, rootDirectory}) => {
 
   const newDockerfile = pm.lockfile
     ? dockerfile.replace(
-        new RegExp(escapeRegExp('ADD package.json'), 'g'),
+        new RegExp(escapeRegExp('ADD package.json package-lock.json'), 'g'),
```
Comment (ahuth) [inline remix.init/index.js:59]: Does this even need to be a regex? 🤔

File: Dockerfile (hunk 1 above)
Comment (jinlee93) [reply Dockerfile:12]: Prob outdated question, but can we use `yarn` / `yarn.lock`?

Comment (ahuth) [reply Dockerfile:12]: Yes we can. The `remix.init` script will override this with `yarn.lock` if need be (although there are some other issues with using yarn right now. There's a ticket to address).

**Pull request merged**